Create docs for v1.0.x (#1855)
woodpecker-ci/woodpecker: docs/versioned_docs/version-1.0/10-intro.md

# Welcome to Woodpecker

> Woodpecker is a simple CI engine with great extensibility. It runs your pipelines inside [containers](https://opencontainers.org/), so if you are already using them in your daily workflow, you'll love Woodpecker for sure.

Reviewer: ```suggestion
Woodpecker is a simple CI engine with great extensibility. It focuses on executing pipelines inside [containers](https://opencontainers.org/). If you are already using containers in your daily workflow, you'll for sure love Woodpecker.
```

![woodpecker](woodpecker.png)

## .woodpecker.yml

- Place your pipeline in a file named `.woodpecker.yml` in your repository
- Pipeline steps can be named as you like
- Run any command in the commands section

```yaml
# .woodpecker.yml
steps:
  build:
    image: debian
    commands:
      - echo "This is the build step"
  a-test-step:
    image: debian
    commands:
      - echo "Testing.."
```

### Steps are containers

- Define any container image as context

>   - either use your own and install the needed tools in custom image or

Reviewer: ```suggestion
  - either use your own and install the needed tools in a custom image
```

>   - search for available images that are already tailored for your needs on container registries like [Docker Hub](https://hub.docker.com/search?type=image)

Reviewer: ```suggestion
  - or search for available images that are already tailored for your needs in image registries like [Docker Hub](https://hub.docker.com/search?type=image)
```

> - List the commands that should be executed in your container, in order to build or test your application

Reviewer: ```suggestion
- List the commands that should be executed in the container
```

```diff
steps:
  build:
-   image: debian
+   image: mycompany/image-with-awscli
    commands:
      - aws help
```

### File changes are incremental

- Woodpecker clones the source code in the beginning

> - Changes to files are persisted through steps as the same volume is mounted to all steps

Reviewer: ```suggestion
- File changes are persisted throughout individual steps as the same volume is being mounted in all steps
```
